### ISSUE TITLE
nofastcancel option and default behaviour

### ISSUE THREAD

**Issue by poljak181** [fastbuild/fastbuild#937]
Hi!
According to the documentation when a task fails, by default any tasks running on other threads are aborted. Does it mean, that when I use command with multiple targets and one of targets fails because of error in source code - all other task will be stop before their ending. For example:
```bash
FBuild.exe -config <path_to_config> -noprogress -wait -summary -clean -nounity -dist target1 target2 target3
```
I added error in target 1 and used above command to compile with 1.08 version. Despite on the compiler find error immediately, I have to wait for other targets fully compiled

**Comment by ffulin** [comment]
There are a couple of things going on here.

*how* the build is stopped
When a build is being stopped due to a compile error, FASTBuild will stop starting new tasks and terminate any other running tasks in order to stop the build as quickly as possible. If -nofastcancel is specified, no new tasks will be started (same as before) but instead of terminating running tasks, they will be allowed to complete gracefully. 

*if* the build is stopped
Whether a build should be stopped when any error is encountered is true by default, but can be disabled with -nostoponerror.  If set, every possible task that can be started will be attempted, regardless of how many errors have been encountered.

*multiple targets on command line*
The complexity here comes from a special case whereby specifying multiple args on the command line automatically implies -nostoponerror. This was done originally because without this, a failure in one task would report all items as broken.
`targetA: FAILED
targetB: FAILED'

In retrospect, this was a poor design decision. Instead, it would be better for those tasks to be differentiated in the output and everything else otherwise behave consistently whether there be 1 target or multiple targets specified on the command line:
`targetA: FAILED
targetB: INCOMPLETE'
and for -nostoponerror to never be implicitly added.

Currently, the only work-around for this issues is to create an Alias() in the bff that wraps your mutliple targets and pass that on the command line. This will give the expected "stop building as quickly as possible behavior".

I hope that makes sense. Let me know if I can clarify.

To summarize, the following work is the appropriate resolution for this issue:
 * remove implicit setting of -nostoponerror when there are multiple targets on the cmd line
 * differentiate the output of FAILED vs INCOMPLETE targets

I'll keep this ticket open as a record of this desired work.

**Comment by ffulin** [comment]
I've implemented the changes described above (-nostoponerror is no longer implicit with multiple targets and output differentiates between failures and incomplete tasks) in https://github.com/fastbuild/fastbuild/commit/4fa331fd2b6ec136f0cc16a066ed98a20582b50e

This change will be in the next release (v1.13)

I'll keep this issue open until that version is released.

**Comment by ffulin** [comment]
v1.13 has now been released and includes the fixes/consistency changes mentioned above.
